File: sdks/python/examples/agent_with_privacy_data.py
import os

import rsa

from alith.data import decrypt, encrypt


def main():
    privacy_data = b"Hello, Privacy Data with PGP!"
    password = os.urandom(32).hex()
    (pub_key, priv_key) = rsa.newkeys(3072)
    encrypted_key = rsa.encrypt(password.encode(), pub_key)
    encrypted_data = encrypt(privacy_data, password)
    try:
        decrypted_password = rsa.decrypt(encrypted_key, priv_key).decode()
    except rsa.DecryptionError:
        raise ValueError("RSA Failed")
    assert decrypted_password == password
    decrypted_data = decrypt(encrypted_data, decrypted_password)
    assert decrypted_data == privacy_data
    print("Crypto test successfully!")


main()

File: sdks/node/examples/agent_with_privacy_data.ts
import { decrypt, encrypt } from "alith/data";
import NodeRSA from "node-rsa";

async function main() {
  const privacyData = "Hello, Privacy Data with PGP!";
  const encoder = new TextEncoder();
  const dataUint8Array = encoder.encode(privacyData);
  const password = "securepassword123456789";
  const rsa = new NodeRSA({ b: 3072 });
  const encryptedKey = rsa.encrypt(password);
  console.log("Encrypted Key:", encryptedKey.toString("base64"));
  const encryptedData = await encrypt(dataUint8Array, password);
  console.log("Encrypted Data:", encryptedData.toString());
  const decryptedPassword = rsa.decrypt(encryptedKey);
  console.log("Decrypted Password:", decryptedPassword.toString());
  const decryptedData = await decrypt(
    encryptedData,
    decryptedPassword.toString()
  );
  console.log("Decrypted Data:", decryptedData.toString());
}

await main();
